repos/system_upgrade/cloudlinux/actors/clearpackageconflicts/actor.py

```diff
@@ -8,29 +8,52 @@
 from leapp.tags import DownloadPhaseTag, IPUWorkflowTag
 from leapp.libraries.common.cllaunch import run_on_cloudlinux
 
+
 class ClearPackageConflicts(Actor):
     """
     Remove several python package files manually to resolve conflicts between versions of packages to be upgraded.
     """
 
-    name = 'clear_package_conflicts'
+    name = "clear_package_conflicts"
     consumes = (InstalledRPM,)
     produces = ()
     tags = (DownloadPhaseTag.Before, IPUWorkflowTag)
 
     @run_on_cloudlinux
     def process(self):
-        problem_packages = ["alt-python37-six", "alt-python37-pytz"]
-        problem_packages_installed = any([has_package(InstalledRPM, pkg) for pkg in problem_packages])
+        problem_packages = [
+            "alt-python37-six",
+            "alt-python37-pytz",
+        ]
+
+        problem_packages_installed = False
+        for pkg in problem_packages:
+            if has_package(InstalledRPM, pkg):
+                self.log.debug("Conflicting package {} detected".format(pkg))
+                problem_packages_installed = True
+                break
 
         if problem_packages_installed:
             problem_dirs = [
                 "/opt/alt/python37/lib/python3.7/site-packages/six-1.15.0-py3.7.egg-info",
-                "/opt/alt/python37/lib/python3.7/site-packages/pytz-2017.2-py3.7.egg-info"]
+                "/opt/alt/python37/lib/python3.7/site-packages/pytz-2017.2-py3.7.egg-info",
+            ]
+            problem_files = []
+
             for p_dir in problem_dirs:
                 try:
                     if os.path.isdir(p_dir):
                         shutil.rmtree(p_dir)
+                        self.log.debug("Conflicting directory {} removed".format(p_dir))
```

Comment:
Why do remove dir instead of package?

Reply:
Because once the dir is removed, the package can be upgraded by `yum` without issues - the conflict was caused by the fact that the paths listed are dirs in the older version, and files in the newer one, and it couldn't be resolved automatically. Once the dirs were gone, yum could put new files in their place no problem.

Reply:
`egg-info` is a dir on CL7 and is a file in CL8?